### DISCUSSION TITLE
[Help]: Svelte 5 support

### DISCUSSION THREAD

**Discussion by SilvianAchim** [davidjerleke/embla-carousel#1052] [Help] [answered]
### Which variants of Embla Carousel are relevant to this feature request?

- [ ] embla-carousel **(Core)**
- [ ] embla-carousel-react
- [ ] embla-carousel-vue
- [x] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-fade
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

### Feature description

I tried migrating to Svelte 5, but because I use this, I can't.

### Additional Context

Additional details here...

### Before submitting

- [X] I've made research efforts and searched the documentation
- [X] I've searched for existing issues
- [X] I agree to follow this project's Contributing Guidelines for feature requests

**Comment by davidjerleke** [comment]
@SilvianAchim could you elaborate and explain why it’s not working with Svelte 5?

**Reply by SilvianAchim** [reply]
`npm install
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: embla-carousel-svelte@8.2.1
npm ERR! Found: svelte@5.1.9
npm ERR! node_modules/svelte
npm ERR!   dev svelte@"^5.0.0" from the root project
npm ERR!   peer svelte@"^4.0.0 || ^5.0.0-next.0" from @sveltejs/kit@2.6.1
npm ERR!   node_modules/@sveltejs/kit
npm ERR!     dev @sveltejs/kit@"^2.5.27" from the root project
npm ERR!     peer @sveltejs/kit@"^2.0.0" from @sveltejs/adapter-auto@3.2.5
npm ERR!     node_modules/@sveltejs/adapter-auto
npm ERR!       dev @sveltejs/adapter-auto@"^3.0.0" from the root project
npm ERR!   11 more (@sveltejs/vite-plugin-svelte, ...)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer svelte@"^3.49.0 || ^4.0.0" from embla-carousel-svelte@8.2.1
npm ERR! node_modules/embla-carousel-svelte
npm ERR!   embla-carousel-svelte@"^8.2.1" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: svelte@4.2.19
npm ERR! node_modules/svelte
npm ERR!   peer svelte@"^3.49.0 || ^4.0.0" from embla-carousel-svelte@8.2.1
npm ERR!   node_modules/embla-carousel-svelte
npm ERR!     embla-carousel-svelte@"^8.2.1" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! `

**Reply by SilvianAchim** [reply]
`┌   shadcn-svelte  v1.0.0-next.0
│
◇  Components to install:
│  carousel
│
▲  The following components/hooks already exist:
│  carousel, button
│
◇  Would you like to overwrite all existing components?
│  Yes, overwrite everything
│
◇  Ready to install components and dependencies?
│  Yes
│
◇  carousel installed at src\lib\components\ui\carousel
│
◇  button installed at src\lib\components\ui\button
│
◓  npm: Installing dependencies│
└  Error: Command failed with exit code 1: npm i -D embla-carousel-svelte bits-ui@next svelte-radix
npm WARN idealTree Removing dependencies.embla-carousel-svelte in favor of devDependencies.embla-carousel-svelte
npm WARN idealTree Removing dependencies.svelte-radix in favor of devDependencies.svelte-radix
npm WARN idealTree Removing dependencies.bits-ui in favor of devDependencies.bits-ui
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: svelte-i18n@4.0.0
npm ERR! Found: svelte@5.1.9
npm ERR! node_modules/svelte
npm ERR!   peer svelte@"^4.0.0 || ^5.0.0-next.0" from @sveltejs/kit@2.6.1
npm ERR!   node_modules/@sveltejs/kit
npm ERR!     peer @sveltejs/kit@"^2.0.0" from @sveltejs/adapter-auto@3.2.5
npm ERR!     node_modules/@sveltejs/adapter-auto
npm ERR!       dev @sveltejs/adapter-auto@"^3.0.0" from the root project
npm ERR!     dev @sveltejs/kit@"^2.5.27" from the root project
npm ERR!   peer svelte@"^5.0.0-next.96 || ^5.0.0" from @sveltejs/vite-plugin-svelte@4.0.0
npm ERR!   node_modules/@sveltejs/vite-plugin-svelte
npm ERR!     peer @sveltejs/vite-plugin-svelte@"^3.0.0 || ^4.0.0-next.1" from @sveltejs/kit@2.6.1
npm ERR!     node_modules/@sveltejs/kit
npm ERR!       peer @sveltejs/kit@"^2.0.0" from @sveltejs/adapter-auto@3.2.5
npm ERR!       node_modules/@sveltejs/adapter-auto
npm ERR!         dev @sveltejs/adapter-auto@"^3.0.0" from the root project
npm ERR!       1 more (the root project)
npm ERR!     peer @sveltejs/vite-plugin-svelte@"^4.0.0-next.0||^4.0.0" from @sveltejs/vite-plugin-svelte-inspector@3.0.1
npm ERR!     node_modules/@sveltejs/vite-plugin-svelte-inspector
npm ERR!       @sveltejs/vite-plugin-svelte-inspector@"^3.0.0-next.0||^3.0.0" from @sveltejs/vite-plugin-svelte@4.0.0
npm ERR!     1 more (the root project)
npm ERR!   15 more (@sveltejs/vite-plugin-svelte-inspector, cmdk-sv, ...)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer svelte@"^3 || ^4" from svelte-i18n@4.0.0
npm ERR! node_modules/svelte-i18n
npm ERR!   svelte-i18n@"^4.0.0" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: svelte@4.2.19
npm ERR! node_modules/svelte
npm ERR!   peer svelte@"^3 || ^4" from svelte-i18n@4.0.0
npm ERR!   node_modules/svelte-i18n
npm ERR!     svelte-i18n@"^4.0.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! 
npm ERR! For a full report see:`

**Comment by davidjerleke** [comment, marked as answer]
@SilvianAchim:

- https://github.com/davidjerleke/embla-carousel/discussions/796#discussioncomment-10635437

**Reply by SilvianAchim** [reply]
Thank you!
